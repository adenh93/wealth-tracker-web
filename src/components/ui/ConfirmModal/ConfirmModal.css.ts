import { styled } from '@mui/material/styles'
import { Box } from '@mui/material'

export const ButtonContainer = styled(Box)({
  display: 'flex',
  columnGap: '0.5rem',
  justifyContent: 'right',
})
