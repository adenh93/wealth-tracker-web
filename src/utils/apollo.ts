import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
