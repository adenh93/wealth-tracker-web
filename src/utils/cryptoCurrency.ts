import { CMC_LOGO_URL } from '../constants'

export const getCMCLogoUrl = (id: number): string => `${CMC_LOGO_URL}/${id}.png`
