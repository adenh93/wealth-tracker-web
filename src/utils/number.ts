const defaultOpts: Intl.NumberFormatOptions = {
  maximumFractionDigits: 2,
  notation: 'standard',
  signDisplay: 'never',
}

export const formatNumber = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  new Intl.NumberFormat('en-US', {
    ...opts,
  }).format(value)

export const formatCurrency = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  formatNumber(value, { ...opts, style: 'currency', currency: 'AUD' })

export const formatPercent = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string => formatNumber(value, { ...opts, style: 'percent' })
