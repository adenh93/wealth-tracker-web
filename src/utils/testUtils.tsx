import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
