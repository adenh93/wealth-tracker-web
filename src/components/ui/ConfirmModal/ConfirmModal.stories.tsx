import ConfirmModal, { ConfirmModalProps } from './ConfirmModal'
import { Story } from '@storybook/react'
import { Typography } from '@mui/material'

const config = {
  title: 'Components/UI/ConfirmModal',
  component: ConfirmModal,
}

const Template: Story<ConfirmModalProps> = (args) => (
  <ConfirmModal {...args}>
    <Typography variant="body2" sx={{ mb: 4 }}>
      This is a confirmation modal
    </Typography>
  </ConfirmModal>
)

export const Primary = Template.bind({})

Primary.args = {
  open: true,
  title: 'Confirm Modal',
  disabled: false,
  handleConfirm: () => {},
  handleClose: () => {},
}

export default config
