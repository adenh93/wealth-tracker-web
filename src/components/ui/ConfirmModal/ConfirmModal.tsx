import { FC } from 'react'
import { Box, Button } from '@mui/material'
import { ButtonContainer } from './ConfirmModal.css'

export interface ConfirmModalProps {
  disabled?: boolean
  handleConfirm: (e: any) => void
  handleClose: (e: any) => void
}

const ConfirmModal: FC<ConfirmModalProps> = ({
  disabled = false,
  handleConfirm,
  handleClose,
  children,
}) => (
  <>
    <Box sx={{ mb: 4 }}>{children}</Box>
    <ButtonContainer>
      <Button
        variant="contained"
        disabled={disabled}
        onClick={handleConfirm}
        data-testid="confirm-button"
      >
        Confirm
      </Button>
      <Button
        variant="contained"
        color="secondary"
        onClick={handleClose}
        data-testid="cancel-button"
      >
        Cancel
      </Button>
    </ButtonContainer>
  </>
)

export default ConfirmModal
