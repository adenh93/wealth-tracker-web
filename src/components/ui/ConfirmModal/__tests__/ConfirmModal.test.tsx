import { Simulate } from 'react-dom/test-utils'
import { render } from '../../../../utils/testUtils'
import ConfirmModal from '../ConfirmModal'

describe('ConfirmModal component tests', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
      <ConfirmModal
        open={true}
        title="Test"
        handleConfirm={jest.fn()}
        handleClose={jest.fn()}
      />
    )

    expect(asFragment).toMatchSnapshot()
  })

  test('confirm button calls handleDelete when clicked', async () => {
    const mockHandleDelete = jest.fn()

    const { getByTestId } = render(
      <ConfirmModal
        open={true}
        title="Test"
        handleConfirm={mockHandleDelete}
        handleClose={jest.fn()}
      />
    )

    const confirmButton = getByTestId('confirm-button') as HTMLButtonElement
    Simulate.click(confirmButton)

    expect(mockHandleDelete).toHaveBeenCalled()
  })

  test('cancel button calls handleClose when clicked', () => {
    const mockHandleClose = jest.fn()

    const { getByTestId } = render(
      <ConfirmModal
        open={true}
        title="Test"
        handleConfirm={jest.fn()}
        handleClose={mockHandleClose}
      />
    )

    const cancelButton = getByTestId('cancel-button')
    Simulate.click(cancelButton)

    expect(mockHandleClose).toHaveBeenCalled()
  })

  test('confirm button is disabled if disabled prop is true', async () => {
    const mockHandleDelete = jest.fn()

    const { getByTestId } = render(
      <ConfirmModal
        open={true}
        title="Test"
        disabled={true}
        handleConfirm={mockHandleDelete}
        handleClose={jest.fn()}
      />
    )

    const confirmButton = getByTestId('confirm-button') as HTMLButtonElement

    expect(confirmButton.disabled).toBe(true)
  })
})
