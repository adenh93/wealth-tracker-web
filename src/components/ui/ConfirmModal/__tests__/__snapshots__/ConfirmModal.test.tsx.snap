// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal component tests matches snapshot 1`] = `[Function]`;
